src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -294,6 +306,38 @@ impl TargetInfo {
                 gctx.shell().warn("non-trivial mutual dependency between target-specific configuration and RUSTFLAGS")?;
             }
 
+            let mut supports_std: Option<bool> = None;
+
+            // The '--print=target-spec-json' is an unstable option of rustc, therefore if
+            // rustc is not a 'Prerelease' build, it's safe to assume that we cannot fetch
+            // the target-spec-json. So don't bother trying.
+            if rustc.version.is_prerelease() {
+                let mut target_spec_process = rustc.workspace_process();
+                apply_env_config(gctx, &mut target_spec_process)?;
+                target_spec_process
```

Comment:
>  I would suggest collecting the JSON information in a separate call to cached_output. That would also avoid needing to guess where the JSON boundaries are.

Couldn't we again use `--print=crate-name`, which print `___` as a delimiter again? I know that is quite hacky, though a second call to `rustc` isn't a good thing for cargo startup time, especially for the stabilization of `cargo -Zscript`.

Reply:
I'm not sure what the purpose of adding the delimiter back in would be? Do you mean reverting back to doing this all in a single rustc call?

Reply:
We put this in a second rustc call to avoid RUSTC_BOOTSTRAP being used on the first one. If startup time is a concern we could ensure it only runs if build-std is requested.

Reply:
> If startup time is a concern we could ensure it only runs if build-std is requested.


That would be ideal! build_std can be accessed via `gctx.cli_unstable(). build_std.is_some()` 

https://doc.rust-lang.org/nightly/nightly-rustc/cargo/core/features/struct.CliUnstable.html#structfield.build_std